--- origsrc/libxcb-1.1/src/Makefile.am	2006-11-24 04:42:54.000000000 -0600
+++ src/libxcb-1.1/src/Makefile.am	2008-10-24 12:30:15.489809000 -0500
@@ -108,96 +108,96 @@
 # * If you add an interface, increment current and age and set revision to 0.
 # * If you change or remove an interface, increment current and set revision
 #   and age to 0.
-libxcb_la_LDFLAGS = -version-info 1:0:0
+libxcb_la_LDFLAGS = -version-info 1:0:0 -no-undefined -export-symbols-regex "_?xcb.*"
 
 BUILT_SOURCES = $(EXTSOURCES) $(EXTHEADERS)
 CLEANFILES = $(EXTSOURCES) $(EXTHEADERS) $(EXTENSION_XML)
 
 XCB_LIBS = libxcb.la
 
-libxcb_xlib_la_LDFLAGS = -version-info 0:0:0
+libxcb_xlib_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xlib_la_LIBADD = $(XCB_LIBS)
 libxcb_xlib_la_SOURCES = xcb_xlib.c
 
 # FIXME: find a way to autogenerate this from the XML files.
 
-libxcb_composite_la_LDFLAGS = -version-info 0:0:0
+libxcb_composite_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_composite_la_LIBADD = $(XCB_LIBS)
 libxcb_composite_la_SOURCES = composite.c composite.h
 
-libxcb_damage_la_LDFLAGS = -version-info 0:0:0
+libxcb_damage_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_damage_la_LIBADD = $(XCB_LIBS)
 libxcb_damage_la_SOURCES = damage.c damage.h
 
-libxcb_dpms_la_LDFLAGS = -version-info 0:0:0
+libxcb_dpms_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_dpms_la_LIBADD = $(XCB_LIBS)
 libxcb_dpms_la_SOURCES = dpms.c dpms.h
 
-libxcb_glx_la_LDFLAGS = -version-info 0:0:0
+libxcb_glx_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_glx_la_LIBADD = $(XCB_LIBS)
 libxcb_glx_la_SOURCES = glx.c glx.h
 
-libxcb_randr_la_LDFLAGS = -version-info 0:0:0
+libxcb_randr_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_randr_la_LIBADD = $(XCB_LIBS)
 libxcb_randr_la_SOURCES = randr.c randr.h
 
-libxcb_record_la_LDFLAGS = -version-info 0:0:0
+libxcb_record_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_record_la_LIBADD = $(XCB_LIBS)
 libxcb_record_la_SOURCES = record.c record.h
 
-libxcb_render_la_LDFLAGS = -version-info 0:0:0
+libxcb_render_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_render_la_LIBADD = $(XCB_LIBS)
 libxcb_render_la_SOURCES = render.c render.h
 
-libxcb_res_la_LDFLAGS = -version-info 0:0:0
+libxcb_res_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_res_la_LIBADD = $(XCB_LIBS)
 libxcb_res_la_SOURCES = res.c res.h
 
-libxcb_screensaver_la_LDFLAGS = -version-info 0:0:0
+libxcb_screensaver_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_screensaver_la_LIBADD = $(XCB_LIBS)
 libxcb_screensaver_la_SOURCES = screensaver.c screensaver.h
 
-libxcb_shape_la_LDFLAGS = -version-info 0:0:0
+libxcb_shape_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_shape_la_LIBADD = $(XCB_LIBS)
 libxcb_shape_la_SOURCES = shape.c shape.h
 
-libxcb_shm_la_LDFLAGS = -version-info 0:0:0
+libxcb_shm_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_shm_la_LIBADD = $(XCB_LIBS)
 libxcb_shm_la_SOURCES = shm.c shm.h
 
-libxcb_sync_la_LDFLAGS = -version-info 0:0:0
+libxcb_sync_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_sync_la_LIBADD = $(XCB_LIBS)
 libxcb_sync_la_SOURCES = sync.c sync.h
 
-libxcb_xevie_la_LDFLAGS = -version-info 0:0:0
+libxcb_xevie_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xevie_la_LIBADD = $(XCB_LIBS)
 libxcb_xevie_la_SOURCES = xevie.c xevie.h
 
-libxcb_xf86dri_la_LDFLAGS = -version-info 0:0:0
+libxcb_xf86dri_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xf86dri_la_LIBADD = $(XCB_LIBS)
 libxcb_xf86dri_la_SOURCES = xf86dri.c xf86dri.h
 
-libxcb_xfixes_la_LDFLAGS = -version-info 0:0:0
+libxcb_xfixes_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xfixes_la_LIBADD = $(XCB_LIBS)
 libxcb_xfixes_la_SOURCES = xfixes.c xfixes.h
 
-libxcb_xprint_la_LDFLAGS = -version-info 0:0:0
+libxcb_xprint_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xprint_la_LIBADD = $(XCB_LIBS)
 libxcb_xprint_la_SOURCES = xprint.c xprint.h
 
-libxcb_xtest_la_LDFLAGS = -version-info 0:0:0
+libxcb_xtest_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xtest_la_LIBADD = $(XCB_LIBS)
 libxcb_xtest_la_SOURCES = xtest.c xtest.h
 
-libxcb_xv_la_LDFLAGS = -version-info 0:0:0
+libxcb_xv_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xv_la_LIBADD = $(XCB_LIBS)
 libxcb_xv_la_SOURCES = xv.c xv.h
 
-libxcb_xvmc_la_LDFLAGS = -version-info 0:0:0
+libxcb_xvmc_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xvmc_la_LIBADD = $(XCB_LIBS)
 libxcb_xvmc_la_SOURCES = xvmc.c xvmc.h
 
-libxcb_xinerama_la_LDFLAGS = -version-info 0:0:0
+libxcb_xinerama_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xinerama_la_LIBADD = $(XCB_LIBS)
 libxcb_xinerama_la_SOURCES = xinerama.c xinerama.h
 
