--- origsrc/libxcb-1.2/src/Makefile.am	2009-02-25 21:31:10.076125000 -0600
+++ src/libxcb-1.2/src/Makefile.am	2009-02-25 21:32:55.013625000 -0600
@@ -24,7 +24,7 @@ nodist_libxcb_la_SOURCES = xproto.c bigr
 # * If you add an interface, increment current and age and set revision to 0.
 # * If you change or remove an interface, increment current and set revision
 #   and age to 0.
-libxcb_la_LDFLAGS = -version-info 2:0:1
+libxcb_la_LDFLAGS = -version-info 2:0:1 -no-undefined -export-symbols-regex "_?xcb.*"
 
 XCB_LIBS = libxcb.la
 
@@ -35,7 +35,7 @@ EXTSOURCES += composite.c
 EXTENSION_XML += composite.xml
 if BUILD_COMPOSITE
 lib_LTLIBRARIES += libxcb-composite.la
-libxcb_composite_la_LDFLAGS = -version-info 0:0:0
+libxcb_composite_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_composite_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_composite_la_SOURCES = composite.c composite.h
 endif
@@ -45,7 +45,7 @@ EXTSOURCES += damage.c
 EXTENSION_XML += damage.xml
 if BUILD_DAMAGE
 lib_LTLIBRARIES += libxcb-damage.la
-libxcb_damage_la_LDFLAGS = -version-info 0:0:0
+libxcb_damage_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_damage_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_damage_la_SOURCES = damage.c damage.h
 endif
@@ -55,7 +55,7 @@ EXTSOURCES += dpms.c
 EXTENSION_XML += dpms.xml
 if BUILD_DPMS
 lib_LTLIBRARIES += libxcb-dpms.la
-libxcb_dpms_la_LDFLAGS = -version-info 0:0:0
+libxcb_dpms_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_dpms_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_dpms_la_SOURCES = dpms.c dpms.h
 endif
@@ -65,7 +65,7 @@ EXTSOURCES += glx.c
 EXTENSION_XML += glx.xml
 if BUILD_GLX
 lib_LTLIBRARIES += libxcb-glx.la
-libxcb_glx_la_LDFLAGS = -version-info 0:0:0
+libxcb_glx_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_glx_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_glx_la_SOURCES = glx.c glx.h
 endif
@@ -75,7 +75,7 @@ EXTSOURCES += randr.c
 EXTENSION_XML += randr.xml
 if BUILD_RANDR
 lib_LTLIBRARIES += libxcb-randr.la
-libxcb_randr_la_LDFLAGS = -version-info 0:0:0
+libxcb_randr_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_randr_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_randr_la_SOURCES = randr.c randr.h
 endif
@@ -85,7 +85,7 @@ EXTSOURCES += record.c
 EXTENSION_XML += record.xml
 if BUILD_RECORD
 lib_LTLIBRARIES += libxcb-record.la
-libxcb_record_la_LDFLAGS = -version-info 0:0:0
+libxcb_record_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_record_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_record_la_SOURCES = record.c record.h
 endif
@@ -95,7 +95,7 @@ EXTSOURCES += render.c
 EXTENSION_XML += render.xml
 if BUILD_RENDER
 lib_LTLIBRARIES += libxcb-render.la
-libxcb_render_la_LDFLAGS = -version-info 0:0:0
+libxcb_render_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_render_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_render_la_SOURCES = render.c render.h
 endif
@@ -105,7 +105,7 @@ EXTSOURCES += res.c
 EXTENSION_XML += res.xml
 if BUILD_RESOURCE
 lib_LTLIBRARIES += libxcb-res.la
-libxcb_res_la_LDFLAGS = -version-info 0:0:0
+libxcb_res_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_res_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_res_la_SOURCES = res.c res.h
 endif
@@ -115,7 +115,7 @@ EXTSOURCES += screensaver.c
 EXTENSION_XML += screensaver.xml
 if BUILD_SCREENSAVER
 lib_LTLIBRARIES += libxcb-screensaver.la
-libxcb_screensaver_la_LDFLAGS = -version-info 0:0:0
+libxcb_screensaver_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_screensaver_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_screensaver_la_SOURCES = screensaver.c screensaver.h
 endif
@@ -125,7 +125,7 @@ EXTSOURCES += shape.c
 EXTENSION_XML += shape.xml
 if BUILD_SHAPE
 lib_LTLIBRARIES += libxcb-shape.la
-libxcb_shape_la_LDFLAGS = -version-info 0:0:0
+libxcb_shape_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_shape_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_shape_la_SOURCES = shape.c shape.h
 endif
@@ -135,7 +135,7 @@ EXTSOURCES += shm.c
 EXTENSION_XML += shm.xml
 if BUILD_SHM
 lib_LTLIBRARIES += libxcb-shm.la
-libxcb_shm_la_LDFLAGS = -version-info 0:0:0
+libxcb_shm_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_shm_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_shm_la_SOURCES = shm.c shm.h
 endif
@@ -145,7 +145,7 @@ EXTSOURCES += sync.c
 EXTENSION_XML += sync.xml
 if BUILD_SYNC
 lib_LTLIBRARIES += libxcb-sync.la
-libxcb_sync_la_LDFLAGS = -version-info 0:0:0
+libxcb_sync_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_sync_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_sync_la_SOURCES = sync.c sync.h
 endif
@@ -155,7 +155,7 @@ EXTSOURCES += xevie.c
 EXTENSION_XML += xevie.xml
 if BUILD_XEVIE
 lib_LTLIBRARIES += libxcb-xevie.la
-libxcb_xevie_la_LDFLAGS = -version-info 0:0:0
+libxcb_xevie_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xevie_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xevie_la_SOURCES = xevie.c xevie.h
 endif
@@ -165,7 +165,7 @@ EXTSOURCES += xf86dri.c
 EXTENSION_XML += xf86dri.xml
 if BUILD_XFREE86_DRI
 lib_LTLIBRARIES += libxcb-xf86dri.la
-libxcb_xf86dri_la_LDFLAGS = -version-info 0:0:0
+libxcb_xf86dri_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xf86dri_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xf86dri_la_SOURCES = xf86dri.c xf86dri.h
 endif
@@ -175,7 +175,7 @@ EXTSOURCES += xfixes.c
 EXTENSION_XML += xfixes.xml
 if BUILD_XFIXES
 lib_LTLIBRARIES += libxcb-xfixes.la
-libxcb_xfixes_la_LDFLAGS = -version-info 0:0:0
+libxcb_xfixes_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xfixes_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xfixes_la_SOURCES = xfixes.c xfixes.h
 endif
@@ -185,7 +185,7 @@ EXTSOURCES += xinerama.c
 EXTENSION_XML += xinerama.xml
 if BUILD_XINERAMA
 lib_LTLIBRARIES += libxcb-xinerama.la
-libxcb_xinerama_la_LDFLAGS = -version-info 0:0:0
+libxcb_xinerama_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xinerama_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xinerama_la_SOURCES = xinerama.c xinerama.h
 endif
@@ -195,7 +195,7 @@ EXTSOURCES += xinput.c
 EXTENSION_XML += xinput.xml
 if BUILD_XINPUT
 lib_LTLIBRARIES += libxcb-xinput.la
-libxcb_xinput_la_LDFLAGS = -version-info 0:0:0
+libxcb_xinput_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xinput_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xinput_la_SOURCES = xinput.c xinput.h
 endif
@@ -205,7 +205,7 @@ EXTSOURCES += xprint.c
 EXTENSION_XML += xprint.xml
 if BUILD_XPRINT
 lib_LTLIBRARIES += libxcb-xprint.la
-libxcb_xprint_la_LDFLAGS = -version-info 0:0:0
+libxcb_xprint_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xprint_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xprint_la_SOURCES = xprint.c xprint.h
 endif
@@ -215,7 +215,7 @@ EXTSOURCES += xselinux.c
 EXTENSION_XML += xselinux.xml
 if BUILD_SELINUX
 lib_LTLIBRARIES += libxcb-xselinux.la
-libxcb_xselinux_la_LDFLAGS = -version-info 0:0:0
+libxcb_xselinux_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xselinux_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xselinux_la_SOURCES = xselinux.c xselinux.h
 endif
@@ -225,7 +225,7 @@ EXTSOURCES += xtest.c
 EXTENSION_XML += xtest.xml
 if BUILD_XTEST
 lib_LTLIBRARIES += libxcb-xtest.la
-libxcb_xtest_la_LDFLAGS = -version-info 0:0:0
+libxcb_xtest_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xtest_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xtest_la_SOURCES = xtest.c xtest.h
 endif
@@ -235,7 +235,7 @@ EXTSOURCES += xv.c
 EXTENSION_XML += xv.xml
 if BUILD_XV
 lib_LTLIBRARIES += libxcb-xv.la
-libxcb_xv_la_LDFLAGS = -version-info 0:0:0
+libxcb_xv_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xv_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xv_la_SOURCES = xv.c xv.h
 endif
@@ -245,7 +245,7 @@ EXTSOURCES += xvmc.c
 EXTENSION_XML += xvmc.xml
 if BUILD_XVMC
 lib_LTLIBRARIES += libxcb-xvmc.la
-libxcb_xvmc_la_LDFLAGS = -version-info 0:0:0
+libxcb_xvmc_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_xvmc_la_LIBADD = $(XCB_LIBS)
 nodist_libxcb_xvmc_la_SOURCES = xvmc.c xvmc.h
 endif
